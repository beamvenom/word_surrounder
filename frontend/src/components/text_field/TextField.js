import styled from 'styled-components'

export const TextFieldText = styled.pre`
  text-align: center;
`
