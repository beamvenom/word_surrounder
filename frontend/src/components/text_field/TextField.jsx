import React from 'react'

const processFile = (file) => {
  file.text().then((res) =>
    fetch('http://localhost:8000/uploadfile/', {
      method: 'POST',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify({ text: res }),
    }).then((response) => {
      response.json().then((value) => {
        document.getElementById('showText').innerHTML = value
        document.getElementById('uploadText').innerHTML = 'Success!'
      })
    }).catch(() => {
      document.getElementById('showText').innerHTML = 'Failed to connect to server'
      document.getElementById('uploadText').innerHTML = 'Failure'
    })
  )
}
const TextField = (props) => {
  if (props.uploadedFile !== null) {
    processFile(props.uploadedFile)
  }
  return (
    <pre data-testid = "showText" id="showText" align="center">
      Processed text will be printed here!
    </pre>
  )
}
export default TextField
