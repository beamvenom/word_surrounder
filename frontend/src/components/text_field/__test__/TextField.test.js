import React from 'react'
import { render, screen } from '@testing-library/react'
import TextField from '../TextField.jsx'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ text: "hej"}),
  })
);
beforeEach(() => {
    render(<TextField uploadedFile = {null}></TextField>)
    fetch.mockClear()
  });

test("Component renders correctly",() =>{
    expect(screen.getByTestId("showText")).toHaveTextContent("Processed text will be printed here!")
})
test("matches snapshot", () => {
  const tree = renderer.create(<TextField uploadedFile = {null}></TextField>)
  expect(tree).toMatchSnapshot()
})
// TODO: test("Process file works correctly",() =>{