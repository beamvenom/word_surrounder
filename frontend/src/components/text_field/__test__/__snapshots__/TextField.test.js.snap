// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<pre
  align="center"
  data-testid="showText"
  id="showText"
>
  Processed text will be printed here!
</pre>
`;
